### PR TITLE
Fix/underscore in response

### PR DESCRIPTION
**Pull request by Brandhoej**
## Description

The endpoints did not conform to the specification, and use case responses contained objects directly from the domain which could cause secrets to leak to the application layer which should only contain information which could be publicly availabe.

### Type of change
It is a bug fix and a security fix. As the HTTP endpoints changed it is also a breaking change.
- [x] Bug fix (non-breaking change which fixes an issue)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### How Has This Been Tested?
The existing tests were altered to conform to the correct specification and additional assertions have been created to catch non-conformance with the Swagger endpoint specification.

### Checklist:

- [X] You have added a clear title and description for the PR
- [X] The CI pass
- [X] There are no merge conflicts
- [X] The code satisfies the [code standards](https://docs.google.com/document/d/1R9n9MeuRwz3WSrnWZUjrbmGeosTlu1Xz8f5F7gu2AJE/edit#heading=h.e9xl4xfoldow) 
- [X] I have added docstrings for my code, where needed
- [X] Code implements intended functionality
- [X] All relevant tests pass (Black, White, Integration, Monkey, etc.)

### CODE REVIEW

File: src/application/usecases/get_an_exercise_use_case.spec.ts [hunk 1]
```diff
@@ -1,28 +1,41 @@
...
+    const exercise = new Exercise(uuidv4(), 'title', 'desc', []);
     const exerciseUseCase = new GetAnExerciseUseCase(repository);
+    const expected: IGetAnExerciseResponse = {
+      exercise: {
+        id: exercise.id,
+        title: exercise.title,
+        description: exercise.description,
+        queries: exercise.queries.map((query) => {
+          return { query: query.query };
+        }),
```
Comment (JakobFaarGreg) [inline src/application/usecases/get_an_exercise_use_case.spec.ts:22]: Det her burde vi bruge ... (spread) operatoren til

File: src/application/usecases/get_all_exercises_use_case.ts [hunk 2]
```diff
@@ -21,7 +28,16 @@ export class GetAllExercisesUseCase
...
-      exercises: await this.repository.getExercises(),
+      exercises: (await this.repository.getExercises()).map((exercise) => {
+        return {
+          id: exercise.id,
+          title: exercise.title,
+          description: exercise.description,
+          queries: exercise.queries.map((query) => {
+            return { query: query.query };
+          }),
+        };
```
Comment (JakobFaarGreg) [inline src/application/usecases/get_all_exercises_use_case.ts:39]: Det her burde vi bruge ... (spread) operatoren til

File: src/presentation/exercise_controller.spec.ts [hunk 3]
```diff
@@ -1,12 +1,9 @@
...
 import { mock } from 'jest-mock-extended';
 import { exerciseController } from './exercise_controller';
-import { Exercise, IExerciseRepository } from '../domain';
+import { Exercise, IExerciseRepository, Query } from '../domain';
 import { v4 as uuidv4 } from 'uuid';
-import {
-  IGetAnExerciseResponse,
-  IGetAllExercisesResponse,
-} from '../application';
+import { IGetAnExerciseResponse } from '../application';
```
Comment (JakobFaarGreg) [inline src/presentation/exercise_controller.spec.ts:6]: Måske vi skal overveje en code convention ift. absolutte stier til imports før relative stier til imports? hvad tænker du?

**Review by Dieuer — APPROVED** [review]
really good job Andreas

**Review by JakobFaarGreg — COMMENTED** [review]
Der skal lige fikses linter fejlen i get_all_exercises_use_case_spec.

Så er der også en 'Forbidden non-null assertion' fejl i index.ts. Jeg tror den kan fikses ved at tilføje `|| envResult.error != null` eller bare ved `!envResult.error`

File: src/application/usecases/get_all_exercises_use_case.ts (hunk 2 above)
Comment (Brandhoej) [reply src/application/usecases/get_all_exercises_use_case.ts:39]: Nej. Spred operatoren bliver brugt til at folde et objekt ud. Vi vil heller ikke have type dependecies med domain oppe i application da det kan medføre en information leak.

File: src/presentation/exercise_controller.spec.ts (hunk 3 above)
Comment (Brandhoej) [reply src/presentation/exercise_controller.spec.ts:6]: https://github.com/OnlineTrainingPlatform/Exercises/pull/6#discussion_r1018989579

**Comment by Brandhoej** [comment]
@JakobFaarGreg Jeg kan ikke en linting fejl kun warnings.

File: src/application/usecases/get_an_exercise_use_case.spec.ts (hunk 1 above)
Comment (Brandhoej) [reply src/application/usecases/get_an_exercise_use_case.spec.ts:22]: https://github.com/OnlineTrainingPlatform/Exercises/pull/6#discussion_r1018989579

Comment (Brandhoej) [reply src/application/usecases/get_an_exercise_use_case.spec.ts:22]: https://github.com/OnlineTrainingPlatform/Exercises/pull/6#discussion_r1018989579

File: src/presentation/exercise_controller.spec.ts (hunk 3 above)
Comment (Brandhoej) [reply src/presentation/exercise_controller.spec.ts:6]: Umiddelbart så synes jeg at absolutte imports (altså fra node_modules skal være i starten) og relative til sidst. Men det er non blocking for PR nu.

**Review by JakobFaarGreg — APPROVED** [review]

**Pull request merged**